common/scala/src/main/scala/org/apache/openwhisk/http/PoolingRestClient.scala

```diff
@@ -110,9 +112,17 @@ class PoolingRestClient(
       if (response.status.isSuccess) {
         Unmarshal(response.entity.withoutSizeLimit).to[T].map(Right.apply)
       } else {
-        // This is important, as it drains the entity stream.
-        // Otherwise the connection stays open and the pool dries up.
-        response.discardEntityBytes().future.map(_ => Left(response.status))
+        Unmarshal(response.entity).to[JsObject].flatMap { responseEntity =>
+          val error = responseEntity.fields.get("error").getOrElse("").toString
+          val reason = responseEntity.fields.get("reason").getOrElse("").toString
```

Comment:
Can you encode this schema into a type? If both of them are optional it'll be like:

```scala
case class Foo(error: Option[String], reason: Option[String])
```

You can then use that in the `Unmarshal(...).to[Foo]` and use the unmarshalled entity directly rather than relying on an implicit schema.

Reply:
Thinking about it, there is no general schema to use here, right? Could we just unmarshal the body as a string and append that somewhere?